mmdet/core/hook/randon_size_hook.py

```diff
@@ -0,0 +1,49 @@
+from mmcv.runner.hooks import HOOKS, Hook
+import random
+from mmcv.runner import get_dist_info
+
+import torch
+from torch import distributed as dist
+
+
+def random_resize(random_size, data_loader, rank, is_distributed, input_size):
+    tensor = torch.LongTensor(2).cuda()
+
+    if rank == 0:
+        size_factor = input_size[1] * 1. / input_size[0]
+        size = random.randint(*random_size)
+        size = (int(32 * size), 32 * int(size * size_factor))
+        tensor[0] = size[0]
+        tensor[1] = size[1]
+
+    if is_distributed:
+        dist.barrier()
+        dist.broadcast(tensor, 0)
+
+    data_loader.dataset.dynamic_scale = (tensor[0].item(), tensor[1].item())
+    return data_loader.dataset.dynamic_scale
+
+
+@HOOKS.register_module()
+class RandomSizeHook(Hook):
+    """Change the image size, currently used in YOLOX.
```

Comment:
Change and synchronize the random image size across ranks, currently used in YOLOX

Reply:
Done as suggested!